katunold/express-sms-management-api
# 167938473: signup new user

#### What does this PR do?
- Implement signup functionality
- implement contact generation logic
- add user input validation
#### Description of Task to be completed?
- A user should be able to signup with the system given he/she submits his/her email, username
and password and in return, the user is given a unique contact number he will use while sending messages to people on the system.
#### How should this be manually tested?
1. Clone the repo using `git clone https://github.com/katunold/express-sms-management-api.git`
2. Check out to this branch using `git checkout ft-signup-new-user-167938473`
3. Install all dependencies using `npm i`
4. Run the app using `npm start`
5. Open postman and paste this url `http://localhost:3000/api/v1/signup` and set the method to post.
6. Add signup data to the payload
```
{
	"email": "user@gmail.com",
	"userName": "User",
	"password": "password123"
}
```
7. Click send and expect to get the following response
```
{
    "message": "You are successfully registered with us ",
    "contact": "0734776210"
}
```
#### What are the relevant pivotal tracker stories?
[167938473](https://www.pivotaltracker.com/story/show/167938473)

#### Postman Documentation
https://www.getpostman.com/collections/64ddd8260bccae82e9f5
